cassiopeia/solver/GreedySolver.py

```diff
@@ -47,50 +48,38 @@ class GreedySolver(CassiopeiaSolver.CassiopeiaSolver):
     """
 
     def __init__(
-        self,
-        character_matrix: pd.DataFrame,
-        missing_char: int,
-        meta_data: Optional[pd.DataFrame] = None,
-        priors: Optional[Dict[int, Dict[int, float]]] = None,
-        prior_transformation: str = "negative_log",
+        self
     ):
 
-        super().__init__(character_matrix, missing_char, meta_data, priors)
-
-        self.weights = None
-
-        if priors:
-            self.weights = solver_utilities.transform_priors(
-                priors, prior_transformation
-            )
-
-        self.unique_character_matrix = self.character_matrix.drop_duplicates()
-
-        self.duplicate_groups = (
-            character_matrix[character_matrix.duplicated(keep=False) == True]
-            .reset_index()
-            .groupby(character_matrix.columns.tolist())["index"]
-            .agg(["first", tuple])
-            .set_index("first")["tuple"]
-            .to_dict()
-        )
+        super().__init__()
 
     def perform_split(
         self,
-        samples: List[str],
-    ) -> Tuple[List[str], List[str]]:
+        character_matrix: pd.DataFrame,
+        samples: List[int],
+        weights: Optional[Dict[int, Dict[int, float]]] = None,
+        missing_state_indicator: int = -1,
+    ) -> Tuple[List[Union[int, str]], List[Union[int, str]]]:
         """Performs a partition of the samples.
 
         Args:
-            samples: A list of samples, represented by their names in the
-                original character matrix
+            character_matrix: Character matrix
+            mutation_frequencies: A dictionary containing the frequencies of
+                each character/state pair that appear in the character matrix
+                restricted to the sample set
+            samples: A list of samples to partition
+            weights: Weighting of each (character, state) pair. Typically a
+                transformation of the priors.
+            missing_state_indicator: Character representing missing data.
 
         Returns:
             A tuple of lists, representing the left and right partition groups
         """
         pass
 
-    def solve(self) -> nx.DiGraph:
+    def solve(self,
+            cassiopeia_tree: CassiopeiaTree,
+            prior_transformation: str = "negative_log"):
```

Comment:
Same as for `DistanceSolver`, but in this case it seems clearer to me: why not have `prior_transformation` be part of `__init__`? It is a hyperparameter that is not `cassiopeia_tree`-dependent.

Reply:
I touched on this in a previous response, but I feel that it's too data-specific to be married to a `CassiopeiaSolver`. But truly, I can see the merit of it being part of the `__init__` and would be happy to put it there if you feel strongly.

Reply:
I touched on this in a previous reply, but I felt that `prior_transformation` was too data-specific to be married to a solver. But truly I see the merit of having it be part of the `__init__` so I'm happy to put it there if you feel strongly.